### PR TITLE
refactor: clean-up build situation and make cali/mcp-server ESM modules

### PR DESCRIPTION
**Pull request by grabbou**
Changelog:
Do not use Bun to run. 
Use Node (with import --tsx) instead to make sure same behavior across dev/prod
Update TypeScript config to get early warnings about incorrect ESM/CJS imports
Use customCondition to point TypeScript to "source" so it gives correct type information without having to rebuild
Use RSLib to generate bundle for each package (generally most reliable after testing tsup and pkg roll)
Do not bundle anything into Cali but AI package that we had to temporarily patch

TL;DR

> React Native CLI is an ESModule converted to CJS. It uses default exports and is transpiled with Babel. When this gets loaded into Bun, Bun follows Babel interop and will make import default resolve properly. However, in production, Node uses different approach. Importing default will import entire module, and to get actual "default" we must call "default.default". More can be read on this matter here https://esbuild.github.io/content-types/#default-interop

### CODE REVIEW

**Comment by grabbou** [comment]
TBH, now that we have `rslib` and it supports watch mode, we should consider dropping Bun to make sure that we run into those issues earlier than after releasing.

**Review by okwasniewski — APPROVED** [review]
